accounts-db/src/accounts_cache.rs

```diff
@@ -233,10 +233,7 @@ impl AccountsCache {
     }
 
     pub fn add_root(&self, root: Slot) {
-        let max_flushed_root = self.fetch_max_flush_root();
-        if root > max_flushed_root || (root == max_flushed_root && root == 0) {
-            self.maybe_unflushed_roots.write().unwrap().insert(root);
-        }
+        self.maybe_unflushed_roots.write().unwrap().insert(root);
```

Comment:
Since we can prove that slot > max_flush_root, we don't need to check root > max_flush_root.
And since we remove the slot > max_flush_root check, there is no need to special case `root == 0` for genesis block too.

Reply:
> we can prove that slot > max_flush_root

Is this because roots are monotonically increasing?

Reply:
yes, exactly!